projects/simple/tests/Tests/Example/Project.hs

```diff
@@ -1,26 +1,39 @@
 module Tests.Example.Project where
 
-import Prelude
+import Clash.Prelude
+import Clash.Prelude.Testbench
 
 import Test.Tasty
 import Test.Tasty.TH
 import Test.Tasty.Hedgehog
+import Clash.Hedgehog.Sized.Unsigned
+import Clash.Hedgehog.Sized.Vector
 
-import Hedgehog ((===))
-import qualified Hedgehog as H
-import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog as HH
 import qualified Hedgehog.Range as Range
 
-import Example.Project (plus)
+-- Import the module containing the `accum` function
+import Examples.Accum (accum)
 
-prop_plusIsCommutative :: H.Property
-prop_plusIsCommutative = H.property $ do
-  a <- H.forAll (Gen.integral (Range.linear minBound maxBound))
-  b <- H.forAll (Gen.integral (Range.linear minBound maxBound))
-  plus a b === plus b a
+-- Define a Hedgehog property to test the `accum` function
+prop_accum :: HH.Property
+prop_accum = HH.property $ do
+  -- Generate vector of random unsigned numbers of at least length 1.
+  someVec <- HH.forAll $ genSomeVec @_ @1 (Range.linear 0 100)
+    (genUnsigned Range.linearBounded)
+  -- Check if the output matches the expected result
+  case someVec of
+    SomeVec SNat vec -> do
```

Comment:
I think this is not a good test.

1. I think this code will be bewildering to people relatively new to Haskell. If we ever need such code in a _starter template_ we should document really well it's broad workings and what parts people should change to accomodate their own tests.
2. However, in this case, it is just really overcomplicated in my opinion. The one good thing about `outputVerifier` is that it translates to HDL. But we don't do that! We don't need to do the whole `SomeVec` dance and `outputVerifier` at all, just use `fromList`, `sampleN` and `HH.===`.

Reply:
It occurred to me that I didn't think the test even worked. So I tried to run the test. Your PR is pretty far from even compiling. For example the first thing I'm greeted by is this:

<details>

```
/home/peter/src/clash/experiments/test-small-fixes/tests/Tests/Example/Project.hs:9:1: error:
    Could not find module ‘Clash.Hedgehog.Sized.Unsigned’
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
  |         
9 | import Clash.Hedgehog.Sized.Unsigned
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            
/home/peter/src/clash/experiments/test-small-fixes/tests/Tests/Example/Project.hs:10:1: error:
    Could not find module ‘Clash.Hedgehog.Sized.Vector’
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
   |        
10 | import Clash.Hedgehog.Sized.Vector
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            
/home/peter/src/clash/experiments/test-small-fixes/tests/Tests/Example/Project.hs:16:1: error:
    Could not find module ‘Examples.Accum’
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
   |        
16 | import Examples.Accum (accum)
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
</details>

After fixing all the errors, the test completed succesfully. But did it actually test anything? I went ahead and deliberately broke `accum`:
```haskell
accum ::
  forall dom .
  HiddenClockResetEnable dom =>
  Signal dom (Unsigned 8) ->
  Signal dom (Unsigned 8)
accum inp = mux (fmap (== maxBound) count) 0 $ mealy accumT 0 inp
 where
  count :: Signal dom (Index 5)
  count = register 0 $ satSucc SatBound <$> count
  accumT s i = (s + i, s)
```
(side-note: in order to write the type declaration for `count`, I had to add `forall`. Of course, I could avoid it by choosing a different way to specify the type, but if it had been monomorphic, I wouldn't have tripped over this. The error message if you forget the `forall` is not beginner-friendly)

And indeed the test still succeeds. It logs a lot of output to the terminal, but it succeeds. Something like this:
```
[...]
cycle(<Clock: System>): 47, outputVerifier
expected value: 109, not equal to actual value: 0

cycle(<Clock: System>): 48, outputVerifier
expected value: 74, not equal to actual value: 0

cycle(<Clock: System>): 49, outputVerifier
expected value: 61, not equal to actual value: 0

cycle(<Clock: System>): 50, outputVerifier
expected value: 58, not equal to actual value: 0
OK (0.03s)
        ✓ <interactive> passed 100 tests.

All 1 tests passed (0.03s)
```

Why is this? Because Hedgehog only asserts that at some point, the test ends. Whether that is that it ends in shambles or succeeds succesfully.